src/backend/gpopt/translate/CTranslatorQueryToDXL.cpp

```diff
@@ -4487,10 +4487,10 @@ CTranslatorQueryToDXL::CreateDXLProjectNullsForGroupingSets(
 
 		INT sort_group_ref = INT(target_entry->ressortgroupref);
 
-		GPOS_ASSERT_IMP(
-			nullptr == sort_grouping_col_mapping,
-			nullptr != sort_grouping_col_mapping->Find(&sort_group_ref) &&
-				"Grouping column with no mapping");
+		GPOS_ASSERT(
+			nullptr != sort_grouping_col_mapping ||
+			(nullptr != sort_grouping_col_mapping->Find(&sort_group_ref) &&
```

Comment:
The condition was rewritten correctly, but if sort_grouping_col_mapping is nullptr, then segfault happens